src/CustomMissionLog.as

```diff
@@ -438,8 +475,9 @@ class CustomMissionLog
 		// 0 - all descriptions
 		// 1 - skip main story
 		// 2 - skip all
+		// NOTE: Assignment mission in SWL don't have mission desc, so always skips them
 		var val = valDesc.GetValue(); 
-		if (val == 0 || (val == 1 && quest.m_MissionType != _global.Enums.MainQuestType.e_Story))
+		if (val == 0 || (val == 1 && quest.m_MissionType != _global.Enums.MainQuestType.e_Story) && !(isSWL && quest.m_MissionType == _global.Enums.MainQuestType["e_AreaMission"]))

```

Comment:
This looks a bit fishy.

`if (A || B && C)` - I'd prefer you either do `if ((A || B) && C)` or `if (A || (B && C))`, whichever's right. I can never remember the operator precedence.
```
if (val == 0 || 
     (val == 1 && quest.m_MissionType != _global.Enums.MainQuestType.e_Story) &&
     !(isSWL && quest.m_MissionType == _global.Enums.MainQuestType["e_AreaMission"]))
```

Reply:
It indeed looks fishy, and I just realized it currently doesn't even work like I wanted to. The && operator has the priority so I need to to use ``((A || B) && C)``